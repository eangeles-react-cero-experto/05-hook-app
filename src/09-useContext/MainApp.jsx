import { Navigate, Route, Routes } from "react-router-dom";
import { AboutPage, HomePage, LoginPage } from "./pages/index.js";
import { Navbar } from "./components/index.js";
import { UserProvider } from "./context/UserProvider.jsx";

export const MainApp = () => {
  return (
    <UserProvider>
      <Navbar />
      <hr />

      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path={"about"} element={<AboutPage />} />
        <Route path={"login"} element={<LoginPage />}></Route>

        <Route path={"*"} element={<Navigate to={"about"} />} />
      </Routes>
    </UserProvider>
  );
};
