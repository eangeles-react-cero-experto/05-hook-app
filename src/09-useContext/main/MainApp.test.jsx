import { render, screen } from "@testing-library/react";
import { MainApp } from "./MainApp.jsx";
import { MemoryRouter } from "react-router-dom";

describe("Test MainApp", () => {
  it("should match with snapshoot", () => {
    const { container } = render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render the HomePage", () => {
    render(
      <MemoryRouter>
        <MainApp />
      </MemoryRouter>,
    );

    expect(screen.getByText("HomePage")).toBeTruthy();
  });

  it("should render the LoginPage", () => {
    render(
      <MemoryRouter initialEntries={["/login"]}>
        <MainApp />
      </MemoryRouter>,
    );

    expect(screen.getByText("LoginPage")).toBeTruthy();
  });

  it("should render the AboutPage", () => {
    render(
      <MemoryRouter initialEntries={["/about"]}>
        <MainApp />
      </MemoryRouter>,
    );

    expect(screen.getByText("AboutPage")).toBeTruthy();
  });
});
