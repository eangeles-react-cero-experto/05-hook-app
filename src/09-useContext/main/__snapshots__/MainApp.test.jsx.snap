// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MainApp should match with snapshoot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark rounded-3"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        useContext
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          class="navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-link active"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link "
            href="/about"
          >
            About
          </a>
          <a
            class="nav-link "
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <hr />
  <h1>
    HomePage 
    <small />
  </h1>
  <hr />
  <pre
    data-testid="code-block"
  />
</div>
`;
